exchanges/request/request.go

```diff
@@ -150,7 +150,20 @@ func (r *Requester) doRequest(ctx context.Context, endpoint EndpointLimit, newRe
 			}
 		}
 
+		reporter := r.reporter
+		if reporter == nil {
+			// no request reporter defined, fallback to global
+			reporter = globalReporter
+		}
```

Comment:
If globalReporter is a fall back, maybe it might be better setting the `Global Reporter` in New() and then override with the requester option? Unless you want to assign and change the global reporter at different times, then it might need a RWMutex for protection on the `globalReporter var`

Reply:
not quite sure if i got your idea, the use-case so far for an application that imports `gct` is to globally define a global reporter that applies to all exchanges that didn't ask for a specific reporter (all so far). We also want this behaviour to be disabled by default so as not to incur in additional request processing.

Reply:
```diff
		@@ -39,6 +39,7 @@ func New(name string, httpRequester *http.Client, opts ...RequesterOption) *Requ
                retryPolicy: DefaultRetryPolicy,
                maxRetries:  MaxRetryAttempts,
                timedLock:   timedmutex.NewTimedMutex(DefaultMutexLockTimeout),
+               reporter:    globalReporter,
        }

        for _, o := range opts {
@@ -150,18 +151,12 @@ func (r *Requester) doRequest(ctx context.Context, endpoint EndpointLimit, newRe
                        }
                }

-               reporter := r.reporter
-               if reporter == nil {
-                       // no request reporter defined, fallback to global
-                       reporter = globalReporter
-               }
-
                start := time.Now()

                resp, err := r.HTTPClient.Do(req)

-               if reporter != nil {
-                       reporter.Latency(r.Name, p.Method, p.Path, time.Since(start))
+               if r.reporter != nil {
+                       r.reporter.Latency(r.Name, p.Method, p.Path, time.Since(start))
                }

                if retry, checkErr := r.retryPolicy(resp, err); checkErr != nil {
```

So add the global reporter to new, which will be overridden by the requester option (if passed in), I am assuming we set up the global report prior to getting a new requester though. If this is not the case then disregard it. But this will just need a !nil comparative instead of var declaration and then checking that and then assigning the global. Let me know what you think.

Reply:
ah right, got it now, that is indeed cleaner.

>. I am assuming we set up the global report prior to getting a new requester though.

that is exactly the case